Mathlib/Data/Sum/Interval.lean

```diff
@@ -363,5 +404,61 @@ lemma Ioo_inr_inr :
     Ioo (inrₗ b₁ : α ⊕ₗ β) (inrₗ b₂) = (Ioo b₁ b₂).map (Embedding.inr.trans toLex.toEmbedding) := by
   rw [← Finset.map_map]; rfl
 
+end LocallyFiniteOrder
+
+section LocallyFiniteOrderBot
+variable [Preorder α] [Preorder β] [Fintype α] [LocallyFiniteOrderBot α] [LocallyFiniteOrderBot β]
+
+instance instLocallyFiniteOrderBot : LocallyFiniteOrderBot (α ⊕ₗ β) where
+  finsetIic := Sum.elim
+    (Iic · |>.map (.trans .inl toLex.toEmbedding))
+    (fun x => Finset.univ.disjSum (Iic x) |>.map toLex.toEmbedding) ∘ ofLex
+  finsetIio := Sum.elim
+    (Iio · |>.map (.trans .inl toLex.toEmbedding))
+    (fun x => Finset.univ.disjSum (Iio x) |>.map toLex.toEmbedding) ∘ ofLex
+  finset_mem_Iic := by simp
+  finset_mem_Iio := by simp
+
+variable (a : α) (b : β)
+
+lemma Iic_inl : Iic (inlₗ a : α ⊕ₗ β) = (Iic a).map (Embedding.inl.trans toLex.toEmbedding) := rfl
+lemma Iic_inr : Iic (inrₗ b : α ⊕ₗ β) = (Finset.univ.disjSum (Iic b)).map toLex.toEmbedding := rfl
+
+lemma Iio_inl : Iio (inlₗ a : α ⊕ₗ β) = (Iio a).map (Embedding.inl.trans toLex.toEmbedding) := rfl
+lemma Iio_inr : Iio (inrₗ b : α ⊕ₗ β) = (Finset.univ.disjSum (Iio b)).map toLex.toEmbedding := rfl
+
+end LocallyFiniteOrderBot
+
+-- TODO: there is a diamond here
```

Comment:
Please indicate what the long term fix is (i.e., that we need to remove that bad instance).